pywbem/cim_operations.py

```diff
@@ -453,11 +453,18 @@ class WBEMConnection(object):
         :class:`~pywbem.BaseOperationRecorder`, each operation
         that is executed on this connection will be recorded by
         invoking its :meth:`~pywbem.BaseOperationRecorder.record` method.
+
+      has_pull_operations (:class:`py:bool`) that defines whether the
+        WBEM server defined by this connection has the pull operations.
+        It can be set by the attribute or if the default of None is set, the
+        first request that uses the pull operations sets the value to True.
+        If set to False, the generator functions will default to using the
+        non-pull equivalent methods
     """
 
     def __init__(self, url, creds=None, default_namespace=DEFAULT_NAMESPACE,
                  x509=None, verify_callback=None, ca_certs=None,
-                 no_verification=False, timeout=None):
+                 no_verification=False, timeout=None, has_pull_operations=None):
```

Comment:
This parameter does not indicate what the server has, but what this connection wants to use, and we even adjust that as we go, to what the server has.

So I suggest renaming this parameter and the corresponding instance variable to `use_pull_operations`.